canonical/dotrun-image
# Dotrun image

## Done
- New Dockerfile:
  - Similar environment to the current dotrun snap, with the same system dependencies
  - Install the latest NodeJS v16
  - Install the latest npm and yarn
  - Copy and install our current dotrun python package
  - Create an ubuntu user in the container and set it as the default user when running the container
- Dotrun python package:
  - Rename Python package to `dotrun-docker` to avoid confusion
  - Removed database / docker-compose logic
  - `dotrun version` changes:
    - Print the `dotrun-docker` version defined in the setup.py instead of env var `SNAP_VERSION`
    - If env var `DOTRUN_VERSION` is present it will print this version as well
- Updated README.md
- Added pull request template
- Update renovate.json

## How to QA
1. Check-out this branch
2. Build this image `docker build . --tag canonicalwebteam/dotrun-image:local`
3. Run this image `docker run -it canonicalwebteam/dotrun-image:local /bin/bash`
4. Run a project
  - `git clone https://github.com/canonical-web-and-design/snapcraft.io/`
  - `cd snapcraft.io`
  - `dotrun`

`dotrun` should install the dependencies and run the project. Once you see this output:
```
Listening at: http://0.0.0.0:8004
```

Visit http://localhost:8004 and check that the website is running.

## Issue / Card
Fixes #23
